crates/router/src/core/webhooks/incoming_v2.rs

```diff
@@ -0,0 +1,673 @@
+                consume_or_trigger_flow,
+                HeaderPayload::default(),
+            ))
+            .await;
+
+            lock_action
+                .free_lock_action(&state, merchant_account.get_id().to_owned())
+                .await?;
+
+            match response {
+                Ok(value) => value,
+                Err(err)
+                    if matches!(
+                        err.current_context(),
+                        &errors::ApiErrorResponse::PaymentNotFound
+                    ) && state
+                        .clone()
+                        .conf
+                        .webhooks
+                        .ignore_error
+                        .payment_not_found
+                        .unwrap_or(true) =>
+                {
+                    metrics::WEBHOOK_PAYMENT_NOT_FOUND.add(
+                        &metrics::CONTEXT,
+                        1,
+                        &add_attributes([("merchant_id", merchant_account.get_id().clone())]),
+                    );
+                    return Ok(WebhookResponseTracker::NoEffect);
+                }
+                error @ Err(_) => error?,
+            }
+        }
+        _ => Err(errors::ApiErrorResponse::WebhookProcessingFailure).attach_printable(
+            "Did not get payment id as object reference id in webhook payments flow",
+        )?,
+    };
+
+    match payments_response {
+        services::ApplicationResponse::JsonWithHeaders((payments_response, _)) => {
+            let payment_id = payments_response.id.clone();
+
+            let status = payments_response.status;
+
+            let event_type: Option<enums::EventType> = payments_response.status.foreign_into();
+
+            // If event is NOT an UnsupportedEvent, trigger Outgoing Webhook
+            if let Some(_outgoing_event_type) = event_type {
+                let _primary_object_created_at = payments_response.created;
+                // TODO: trigger an outgoing webhook to merchant
+                // Box::pin(super::create_event_and_trigger_outgoing_webhook(
+                //     state,
+                //     merchant_account,
+                //     profile,
+                //     &key_store,
+                //     outgoing_event_type,
+                //     enums::EventClass::Payments,
+                //     payment_id.get_string_repr().to_owned(),
+                //     enums::EventObjectType::PaymentDetails,
+                //     api::OutgoingWebhookContent::PaymentDetails(Box::new(payments_response)),
+                //     Some(primary_object_created_at),
+                // ))
+                // .await?;
+            };
+
+            let response = WebhookResponseTracker::Payment { payment_id, status };
+
+            Ok(response)
+        }
+
+        _ => Err(errors::ApiErrorResponse::WebhookProcessingFailure)
+            .attach_printable("received non-json response from payments core")?,
+    }
+}
+
+async fn get_payment_id(
```

Comment:
return objects from the function, just to get a id we shouldn't do database call, if we do so also, return the object so that, the objects can be used later. change the function name too.

Reply:
We are not fetching the object in all the cases, in case of `PaymentIdType::PaymentIntentId` we are directly returning the id

Reply:
Can we make this consistent, we should avoid reading same object twice in the flow. Like in redirection, we eliminated double reads